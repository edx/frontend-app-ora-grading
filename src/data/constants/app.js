import { getConfig } from '@edx/frontend-platform';

export const routePath = `${getConfig().PUBLIC_PATH}:courseId`;
export const locationId = window.location.pathname.slice(1);
