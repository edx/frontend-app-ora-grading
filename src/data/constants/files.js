import { StrictDict } from 'utils';

export const FileTypes = StrictDict({
  pdf: 'pdf',
  jpg: 'jpg',
  jpeg: 'jpeg',
  png: 'png',
  bmp: 'bmp',
});

export default FileTypes;
