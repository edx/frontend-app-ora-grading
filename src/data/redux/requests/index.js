export { actions, reducer } from './reducer';
