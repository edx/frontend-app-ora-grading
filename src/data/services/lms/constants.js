import { StrictDict } from 'utils';

export const lockStatuses = StrictDict({
  unlocked: 'unlocked',
  locked: 'locked',
  inProgress: 'in-progress',
});

export const gradeStatuses = StrictDict({
  ungraded: 'ungraded',
  graded: 'graded',
});

export const gradingStatuses = StrictDict({
  ungraded: gradeStatuses.ungraded,
  graded: gradeStatuses.graded,
  locked: lockStatuses.locked,
  inProgress: lockStatuses.inProgress,
});

export const gradingStatusDisplay = StrictDict({
  [gradingStatuses.ungraded]: 'Ungraded',
  [gradingStatuses.locked]: 'Currently being graded by someone else',
  [gradingStatuses.graded]: 'Grading Complete',
  [gradingStatuses.inProgress]: 'You are currently grading this response',
});

export const feedbackRequirement = StrictDict({
  disabled: 'disabled',
  required: 'required',
  optional: 'optional',
});
