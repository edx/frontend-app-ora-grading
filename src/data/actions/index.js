import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';
import submissions from './submissions';

export default StrictDict({
  app,
  grading,
  submissions,
});
