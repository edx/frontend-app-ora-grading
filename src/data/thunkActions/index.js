import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';

export default StrictDict({
  app,
  grading,
});
