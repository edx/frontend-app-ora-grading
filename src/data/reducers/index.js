import { combineReducers } from 'redux';

import app from './app';
import submissions from './submissions';

/* istanbul ignore next */
const rootReducer = combineReducers({
  app,
  submissions,
});

export default rootReducer;
