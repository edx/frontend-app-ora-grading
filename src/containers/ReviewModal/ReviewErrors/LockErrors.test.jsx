import React from 'react';
import { shallow } from 'enzyme';

import { selectors } from 'data/redux';
import { ErrorStatuses, RequestKeys } from 'data/constants/requests';

import {
  LockErrors,
  mapStateToProps,
} from './LockErrors';

jest.mock('data/redux', () => ({
  selectors: {
    requests: {
      errorStatus: (...args) => ({ errorStatus: args }),
      isFailed: (...args) => ({ isFailed: args }),
    },
  },
}));

let el;
jest.mock('./ReviewError', () => 'ReviewError');

const requestKey = RequestKeys.setLock;

describe('LockErrors component', () => {
  const props = {
    isFailed: true,
  };
  describe('component', () => {
    beforeEach(() => {
      el = shallow(<LockErrors {...props} />);
      el.instance().dismissError = jest.fn().mockName('this.dismissError');
    });
    describe('snapshots', () => {
      test('no failure', () => {
        el.setProps({ isFailed: false });
        expect(el.instance().render()).toMatchSnapshot();
      });
      test('snapshot: error with bad request', () => {
        el.setProps({ errorStatus: ErrorStatuses.badRequest });
        expect(el.instance().render()).toMatchSnapshot();
      });
      test('snapshot: error with conflicted lock', () => {
        el.setProps({ errorStatus: ErrorStatuses.forbidden });
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('errorStatus loads from requests.errorStatus(setLock)', () => {
      expect(mapped.errorStatus).toEqual(
        selectors.requests.errorStatus(testState, { requestKey }),
      );
    });
  });
});
