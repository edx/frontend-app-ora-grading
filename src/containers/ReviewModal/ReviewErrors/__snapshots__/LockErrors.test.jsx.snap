// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LockErrors component component snapshots no failure 1`] = `null`;

exports[`LockErrors component component snapshots snapshot: error with bad request 1`] = `
<ReviewError
  headingMessage={
    Object {
      "defaultMessage": "Invalid request. Please check your input.",
      "description": "Error lock request for missing params",
      "id": "ora-grading.ReviewModal.errorLockBadRequestHeading",
    }
  }
>
  <FormattedMessage
    defaultMessage="Invalid request. Please check your input."
    description="Error lock request for missing params"
    id="ora-grading.ReviewModal.errorLockBadRequest"
  />
</ReviewError>
`;

exports[`LockErrors component component snapshots snapshot: error with conflicted lock 1`] = `
<ReviewError
  headingMessage={
    Object {
      "defaultMessage": "The lock owned by another user",
      "description": "Error lock by someone else",
      "id": "ora-grading.ReviewModal.errorLockContestedHeading",
    }
  }
>
  <FormattedMessage
    defaultMessage="The lock owned by another user"
    description="Error lock by someone else"
    id="ora-grading.ReviewModal.errorLockContested"
  />
</ReviewError>
`;
